docs/design/connection-tester.md

```diff
@@ -0,0 +1,50 @@
+# Connection Tester
+
+The goal of this document is to sketch out possible approaches for implementing _Connection testing_ in the context of iPaaS. This backend concept is introduced to support the use case to quickly test whether a given `Connection`  (which is the combination of a `Connector` and a `ConnectorConfiguration`) is actually usable. The typical use case is, that the user pushes a button "Test connection" on the "Connection Details" page which actually triggers the test of the connection and that can be either successful or failing. When it fails, a detailed error message describes not only the error itself but also how to resolve the error (e.g. "Cannot connect because of invalid credentials"). 
+
+## Communication flow
+
+When the user triggers this test function from within the UI, a backend call to `ipaas-rest` will be issued, which in turn will trigger a (still to be defined) method on the referenced connector with the concrete configuration given. This method, let's call it `ping()`, must be side effect free, which should be possible for most endpoints. The method takes the configuration as argument for the connector and returns a result objects which contains the result of the check (ok / nok) and in case of a failure the error message (question: Should we already think about I18N here ?)
+
+## Side effect free connection test
+
+Endpoints like database or REST endpoints can be checked side-effect free (e.g. for HTTP based services on could try a `HEAD` request to the root context). If this is not possible because e.g. some payload would be needed to be transferred in order to determine whether the connection works, then a connection test in the sense described here can't be performed. Instead some "sample flow run" needs to be performed which moves a test payload through the flow where the connector is at the the end (but this should be also represented differently on the UI).
```

Comment:
Completely agreed, was probably stated a bit misleading. For me the important thing is that the API is uniform over all connectors, without knowledge how the actual check is done. Input is a concrete configuration (which is probable a generic typeless Map, with meaning specific for the connector ?), output is the ping result (ok/nok, error message if nok)